app/templates/main/map.html

```diff
@@ -0,0 +1,72 @@
+<html>
+<head>
+<script src="http://maps.googleapis.com/maps/api/js"></script>
+<script>
+geocoder = new google.maps.Geocoder();
+address = document.getElementById("address");
+function initialize() {
+var initial_coords = new google.maps.LatLng(39.952, -75.195);
+var mapProp = {
+center: initial_coords,
+zoom:17,
+mapTypeId: google.maps.MapTypeId.ROADMAP
+};
+map=new google.maps.Map(document.getElementById("googleMap"),mapProp);
+}
+function reinitialize() {
+var initial_coords = new google.maps.LatLng(39.952, -75.195);
+var mapProp = {
+center: initial_coords,
+zoom:17,
+mapTypeId: google.maps.MapTypeId.ROADMAP
+};
+var map=new google.maps.Map(document.getElementById("googleMap"),mapProp);
+if (geocoder) {
+console.log(address);
+geocoder.geocode( { 'address': address}, function(results, status) {
+if (status == google.maps.GeocoderStatus.OK) {
+if (status != google.maps.GeocoderStatus.ZERO_RESULTS) {
+map.setCenter(results[0].geometry.location);
+}
+else {
+alert("Address not found!");
+}
+}
+else {
+alert("Geocode not successful - " + status);
+}
+});
+}
+}
+function update_center() {
+geocoder = new google.maps.Geocoder();
+address = document.addressForm.address.value;
+if (geocoder) {
+console.log(address);
+geocoder.geocode( { 'address': address}, function(results, status) {
+if (status == google.maps.GeocoderStatus.OK) {
+if (status != google.maps.GeocoderStatus.ZERO_RESULTS) {
+map.setCenter(results[0].geometry.location);
+}
+else {
+alert("Address not found!");
+}
+}
+else {
+alert("Geocode not successful - " + status);
+}
+});
+}
+}
+
+google.maps.event.addDomListener(window, 'load', initialize);
+</script>
+<form name = "addressForm" action="JavaScript:update_center()">
+    <input type="text" name="address" id="address" value="Enter Address">
+    <br><input type="submit" value="Submit"><br>
+</form>
+</head>
+<body>
+<div id="googleMap" style="width:500px;height:380px;"></div>
```

Comment:
anything that uses the `style` tag should be stored in a scss file (since we are using SASS as our css framework) - maybe create a `map.scss` file and put it in `app/assets/styles` containing your styling